driver-core/src/main/com/mongodb/client/model/FullDocument.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2017 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model;
+
+import static java.lang.String.format;
+
+/**
+ *
+ * Change Stream fullDocument configuration.
+ *
+ * <p>Determines what to return when using a Change Stream. Defaults to {@link FullDocument#NONE}.
+ * When set to {@link FullDocument#LOOKUP}, the change stream for partial updates will include both a delta describing the
+ * changes to the document as well as a copy of the entire document that was changed from <em>some time</em> after the change occurred.</p>
+ *
+ * @since 3.6
+ * @mongodb.server.release 3.6
+ */
+public enum FullDocument {
```

Comment:
This probably needs better documentation - at the moment its a little awkward. So any suggestions would be great.